caffe2/python/trt/transform.py

```diff
@@ -0,0 +1,105 @@
+## @package onnx
+#Module caffe2.python.trt.transform
+
+"""
+TensorRT related transformation
+Note that ONNX-TRT enforce an NCHW input!
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from caffe2.proto import caffe2_pb2
+from caffe2.python.onnx.helper import c2_native_run_net, c2_native_run_op
+import caffe2.python.onnx.frontend as c2_front
+import caffe2.python._import_c_extension as C
+import numpy as np
+
+def _dim_values_to_list(dim_values):
+    return [x.dim_value for x in dim_values]
+
+
+def _get_output_shapes(output_value_infos):
+    names = [x.name for x in output_value_infos]
+    shapes = [_dim_values_to_list(x.type.tensor_type.shape.dim) for x in output_value_infos]
+    return dict(zip(names, shapes))
+
+
+def convert_onnx_model_to_trt_op(onnx_model,
+        max_batch_size=50,
+        max_workspace_size=2*1024*1024,
+        verbosity=1,
+        debug_builder=False):
+    """
+    Convert the whole ONNX model to a TensorRT C2 op
+    """
+    trt_str = C.onnx_to_trt_op(onnx_model.SerializeToString(),
```

Comment:
so if compiled without TRT support or without GPU the method wouldn't just exist. So maybe produce a better messages in this case?

Reply:
OK. I'll create stubs for this and error out with clear messages.